File: internal/content/testdata/bench/java/nc-encode-decode-strings.java
import java.util.*;

class Solution {
    private String encode(String[] strs) {
        StringBuilder sb = new StringBuilder();
        for (String s : strs) {
            sb.append(s.length()).append('#').append(s);
        }
        return sb.toString();
    }

    private String[] decode(String s) {
        List<String> res = new ArrayList<>();
        int i = 0;
        while (i < s.length()) {
            int j = i;
            while (s.charAt(j) != '#') j++;
            int length = Integer.parseInt(s.substring(i, j));
            res.add(s.substring(j + 1, j + 1 + length));
            i = j + 1 + length;
        }
        return res.toArray(new String[0]);
    }

    public String[] encode_decode(String[] strs) {
        return decode(encode(strs));
    }
}

File: internal/content/testdata/bench/typescript/nc-encode-decode-strings.ts
function encode(strs: string[]): string {
    return strs.map(s => s.length + '#' + s).join('');
}

function decode(s: string): string[] {
    const res: string[] = [];
    let i = 0;
    while (i < s.length) {
        let j = i;
        while (s[j] !== '#') j++;
        const length = parseInt(s.slice(i, j));
        res.push(s.slice(j + 1, j + 1 + length));
        i = j + 1 + length;
    }
    return res;
}

function encode_decode(strs: string[]): string[] {
    const s = encode(strs);
    return decode(s);
}
